### PR TITLE
fix(auth): clear oauth data before initiating a new oauth flow

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

It will try to clear oauth data before starting a new oauth flow. This prevents the `signInWithRedirect` from hanging if the oauth flow was previously cancelled.

#### Description of how you validated changes
- Created a sample app to test changes 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/providers/cognito/apis/signInWithRedirect.ts [hunk 1]
```diff
@@ -42,6 +42,9 @@ export async function signInWithRedirect(
 	assertTokenProviderConfig(authConfig);
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/signInWithRedirect.ts:42]: It won't let me comment on it directly, but while you're in here could you please address the todo in the customer facing doc string 🙏

**Review by jimblanc — COMMENTED** [review]
Looks good! One comment on docs

File: packages/auth/src/providers/cognito/apis/signInWithRedirect.ts (hunk 1 above)
Comment (israx) [reply packages/auth/src/providers/cognito/apis/signInWithRedirect.ts:43]: done

**Review by jimblanc — APPROVED** [review]

**Review by cshfang — APPROVED** [review]
Could you please just double check that this workflow works as well?
* Try to login via oauth
* Cancel
* Login via user/pass instead of oauth

**Review by cshfang — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**